### PR TITLE
Firebase

### PR DESCRIPTION
**Pull request by joaoachando**

### CODE REVIEW

File: android/app/build.gradle [hunk 1]
```diff
@@ -23,6 +23,7 @@ if (flutterVersionName == null) {
 
 apply plugin: 'com.android.application'
 apply plugin: 'kotlin-android'
+apply plugin: 'com.google.gms.google-services'
```
Comment (andretmarques) [inline android/app/build.gradle:26]: nit: alphabetically

File: android/app/build.gradle [hunk 2]
```diff
@@ -65,4 +66,6 @@ flutter {
 
 dependencies {
     implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
+	implementation platform('com.google.firebase:firebase-bom:29.0.3')
+	implementation 'com.google.firebase:firebase-analytics'
```
Comment (andretmarques) [inline android/app/build.gradle:70]: lint

File: android/build.gradle [hunk 3]
```diff
@@ -8,6 +8,7 @@ buildscript {
     dependencies {
         classpath 'com.android.tools.build:gradle:4.1.0'
         classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
+		classpath 'com.google.gms:google-services:4.3.10'
```
Comment (andretmarques) [inline android/build.gradle:11]: lint

**Review by miguelrnf — APPROVED** [review]

File: lib/main/UniHubApp.dart [hunk 4]
```diff
@@ -1,35 +1,63 @@
...
+                child: CircularProgressIndicator(),
+              );
+            }
+          },
+        ),
+        // initialRoute: '/LoginPage',
+        // routes: {
+        //   '/LandingPage': (context) => const LandingPage(username: username),
+        //   '/LoginPage': (context) => const LoginPage()
+        // },
```
Comment (andretmarques) [inline lib/main/UniHubApp.dart:57]: don't comment routes. just comment initial

File: lib/signin/google/SignInPage.dart [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+class GoogleSignInProvider extends ChangeNotifier {
+  final googleSignIn = GoogleSignIn();
+  GoogleSignInAccount? _user;
+  GoogleSignInAccount get user => _user!;
+
+  Future googleLogin() async {
+    final googleUser = await googleSignIn.signIn();
+    if(googleUser == null) return;
+    _user = googleUser;
```
Comment (andretmarques) [inline lib/signin/google/SignInPage.dart:13]: `_user`?

**Review by andretmarques — CHANGES_REQUESTED** [review]
!!!!

File: lib/signin/google/SignInPage.dart [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+import 'package:google_sign_in/google_sign_in.dart';
+
+class GoogleSignInProvider extends ChangeNotifier {
+  final googleSignIn = GoogleSignIn();
+  GoogleSignInAccount? _user;
+  GoogleSignInAccount get user => _user!;
+
+  Future googleLogin() async {
+    final googleUser = await googleSignIn.signIn();
+    if(googleUser == null) return;
```
Comment (andretmarques) [inline lib/signin/google/SignInPage.dart:12]: lint

**Review by andretmarques — CHANGES_REQUESTED** [review]

**Review by andretmarques — APPROVED** [review]

File: lib/login/LoginPage.dart [hunk 7]
```diff
@@ -41,9 +42,9 @@ class _LoginPageState extends State<LoginPage> {
                   crossAxisAlignment: CrossAxisAlignment.center
               ) ,
               const Padding(padding: EdgeInsets.only(bottom: 50.0)),
-              _buildTextLabel("LOGIN WITH:"),
-              Row(children: [
-                _buildLogos(AssetType.Facebook),
+              utils.buildTextLabel("LOGIN WITH:"),
+              Row(children:
+                [_buildLogos(AssetType.Facebook),
```
Comment (andretmarques) [inline lib/login/LoginPage.dart:47]: !!!

**Review by andretmarques — CHANGES_REQUESTED** [review]

File: lib/login/LoginPage.dart (hunk 7 above)
Comment (andretmarques) [inline lib/login/LoginPage.dart:47]:
```suggestion
              Row(children: [
                _buildLogos(AssetType.Facebook),
```

Comment (andretmarques) [inline lib/login/LoginPage.dart:47]:
```suggestion
              Row(children: [
                _buildLogos(AssetType.Facebook),
```

**Pull request merged**